### PR TITLE
Update contribution document

### PR DESCRIPTION
**Pull request by tigercosmos**
Add more information for installing packages.

### CODE REVIEW

**Review by Dimi1010 — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,11 +22,14 @@ Every code contribution to this project is highly valued and appreciated. I enco
...
-     - For Windows (MSI install): https://github.com/danmar/cppcheck/releases/download/2.7/cppcheck-2.7-x64-Setup.msi
-     - Build from source: https://github.com/danmar/cppcheck/releases/tag/2.7
+   - Make sure that pre-commit hooks are passing by using `pre-commit run --all-files`. For pre-commit hooks you need to install `cppcheck`, `clang-format`, and `ruff` on your system. You can install them using the following commands:
+      - `pre-commit`: `pip install pre-commit`
+      - `cppcheck` (version 2.7 is recommended):
+         - For Linux (apt) `sudo apt install cppcheck`
+         - For Windows (using `choco`) `choco install cppcheck --version=2.7`
+         - For Windows (MSI install): https://github.com/danmar/cppcheck/releases/download/2.7/cppcheck-2.7-x64-Setup.msi
+         - Build from source: https://github.com/danmar/cppcheck/releases/tag/2.7
+      - `clang-format` and `ruff`: `pip install clang-format==18.1.6 ruff==0.6.5`
```
Comment (seladb) [inline CONTRIBUTING.md:32]: Let's wait until we merge [this PR](https://github.com/seladb/PcapPlusPlus/pull/1584) to see if we need it

Comment (tigercosmos) [reply CONTRIBUTING.md:32]: @seladb pre-commit will run ruff in a venv, but I think users will also run ruff without using pre-commit.

Comment (seladb) [reply CONTRIBUTING.md:32]: But we ask users to install `pre-commit` that includes `ruff`, do we really need this redundancy?

Comment (tigercosmos) [reply CONTRIBUTING.md]: Sure, removed ruff here.

**Review by seladb — APPROVED** [review]

**Pull request merged**